### PR TITLE
[IMPROVE] Configuration for indices in Apps-Engine models

### PR DESCRIPTION
**Pull request by d-gubert**
## Proposed changes (including videos or screenshots)

* Add `appId` field to the data saved by the Scheduler
* Add `appId` index to `rocketchat_apps_persistence` model
* Skip "trash collection" when deleting records from `rocketchat_apps_persistence`
* Add a new setting to control for how long we should keep logs from the apps

image

image

## Further comments

I'm not so certain about the setting name and the alert texts, inputs are very welcome :)

### CODE REVIEW

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -463,6 +463,11 @@
   "Apps_License_Message_publicKey": "There has been an error trying to decrypt the license. Please sync your workspace in the Connectivity Services and try again",
   "Apps_License_Message_renewal": "License has expired and needs to be renewed",
   "Apps_License_Message_seats": "License does not have enough seats to accommodate the current amount of active users. Please increase the number of seats",
+  "Apps_Logs_TTL": "Number of days to keep logs from apps stored",
+  "Apps_Logs_TTL_7days": "7 Days",
+  "Apps_Logs_TTL_14days": "14 Days",
+  "Apps_Logs_TTL_30days": "30 Days",
```
Comment (d-gubert) [inline packages/rocketchat-i18n/i18n/en.i18n.json:469]:
```suggestion
  "Apps_Logs_TTL_7days": "7 days",
  "Apps_Logs_TTL_14days": "14 days",
  "Apps_Logs_TTL_30days": "30 days",
```

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -463,6 +463,11 @@
   "Apps_License_Message_publicKey": "There has been an error trying to decrypt the license. Please sync your workspace in the Connectivity Services and try again",
   "Apps_License_Message_renewal": "License has expired and needs to be renewed",
   "Apps_License_Message_seats": "License does not have enough seats to accommodate the current amount of active users. Please increase the number of seats",
+  "Apps_Logs_TTL": "Number of days to keep logs from apps stored",
+  "Apps_Logs_TTL_7days": "7 days",
+  "Apps_Logs_TTL_14days": "14 days",
+  "Apps_Logs_TTL_30days": "30 days",
+  "Apps_Logs_TTL_Alert": "Depending on the size of the Logs collection, changing this setting might cause slowness for some moments",
```
Comment (d-gubert) [inline packages/rocketchat-i18n/i18n/en.i18n.json:470]:
```suggestion
  "Apps_Logs_TTL_Alert": "Depending on the size of the Logs collection, changing this setting may cause slowness for some moments",
```

File: app/models/server/models/apps-persistence-model.js [hunk 3]
```diff
@@ -3,5 +3,12 @@ import { Base } from './_Base';
 export class AppsPersistenceModel extends Base {
 	constructor() {
 		super('apps_persistence');
+
+		this.tryEnsureIndex({ appId: 1 });
+	}
+
+	// Bypass trash collection
```
Comment (thassiov) [inline app/models/server/models/apps-persistence-model.js:10]: to match `apps-logs-model`, right?

**Review by thassiov — APPROVED** [review]
LGTM!

File: app/models/server/models/apps-persistence-model.js (hunk 3 above)
Comment (d-gubert) [reply app/models/server/models/apps-persistence-model.js:10]: Yup 😬

**Pull request merged**